BootloaderCorePkg/Tools/CommonUtility.py

```diff
@@ -209,26 +209,56 @@ def run_process (arg_list, print_cmd = False, capture_out = False):
 
     return output
 
-def rsa_sign_file (priv_key, pub_key, hash_type, in_file, out_file, inc_dat = False, inc_key = False):
+def rsa_sign_file (priv_key, pub_key, hash_type, sign_scheme, in_file, out_file, inc_dat = False, inc_key = False):
+
     _hash_type_string = {
-        "SHA2_256"    : '-sha256',
-        "SHA2_384"    : '-sha384',
-        "SHA2_512"    : '-sha512',
+        "SHA2_256"    : 'sha256',
+        "SHA2_384"    : 'sha384',
+        "SHA2_512"    : 'sha512',
+    }
+
+    _sign_scheme_string = {
+        "RSA_PCKS_1_5"    : 'pkcs1',
+        "RSA_PSS"         : 'pss',
     }
 
     bins = bytearray()
     if inc_dat:
         bins.extend(get_file_data(in_file))
 
-    cmdargs = [get_openssl_path(), 'dgst' , '%s' % _hash_type_string[hash_type], '-sign', '%s' % priv_key,
-               '-out', '%s' % out_file, '%s' % in_file]
+    # Temporary files to store hash generated
+    hash_file_tmp = out_file+'.hash.tmp'
+    hash_file     = out_file+'.hash'
+
+    # Generate hash using openssl dgst in hex format
+    cmdargs = [get_openssl_path(), 'dgst', '-'+'%s' % _hash_type_string[hash_type], '-out', '%s' % hash_file_tmp, '%s' % in_file]
+    run_process (cmdargs)
+
+    #sample hash data generated by dgst command
+    #SHA256()= 7658f28ae53f61b6dbaeac5ea13d0a169a3f325773bfa5c3cdfc2d0cad4a74d3
+
+    # Extract hash form dgst command output and convert to ascii
+    with open(hash_file_tmp, 'r') as fin:
+        hashdata = fin.read()
+    fin.close()
+
+    pos = hashdata.index("=")
+    hashdata = hashdata[pos+2:pos+2+64]

```

Comment:
if '=' is not found?
if the size of hashdata is not 64?
Can you also use try/except?